### PR TITLE
MWPW-142798 + MWPW-142872: Made the height-fit-content class to be set automatically

### PR DESCRIPTION
**Pull request by mirafedas**
When Trial with Purchase pages (e.g. https://www.adobe.com/mini-plans/photoshop.html?mid=ft&amp;web=1&amp;pint=Photoshop ) are rendered inside modals, we need to set up an automatic height adjustment to eliminate the blank space on the bottom of the page. For that the 'height-fit-content' class should be set on the modal dialog element. As adding this class manually would require a significant authoring effort, we now set it automatically in the code for all cases when the URL of the page inside iFrame includes the '/mini-plans/' part together with 'mid=ft' and 'web=1' params. 
This PR also includes the CSS changes from [this PR](https://github.com/adobecom/milo/pull/1911) as they are required for the modal styles to work properly.

Resolves: 
[MWPW-142798](https://jira.corp.adobe.com/browse/MWPW-142798)
[MWPW-142872](https://jira.corp.adobe.com/browse/MWPW-142872)

**Test URLs:**
- Before: https://www.adobe.com/products/illustrator/campaign/pricing.html#modal-illustrator
- After: https://mwpw-142798-autoset-height-fit-content--milo--mirafedas.hlx.live/drafts/mirafedas/twp#mini-plans-web-cta-creative-cloud-card

For testers, please test the following issues:
https://jira.corp.adobe.com/browse/MWPW-142872
https://jira.corp.adobe.com/browse/MWPW-142798
https://jira.corp.adobe.com/browse/MWPW-142399 (check for regression)

### CODE REVIEW

**Review by sharath-kannan — APPROVED** [review]
Looks good.

**Comment by Roycethan** [comment]
@mirafedas I checked your examples given for the fix : 
Without class authored: https://main--cc--adobecom.hlx.live/de/creativecloud/animation/discover?martech=off&milolibs=mwpw-142798-autoset-height-fit-content--milo--mirafedas#animate
With class authored (as per live example): https://mwpw-142798-autoset-height-fit-content--milo--mirafedas.hlx.live/drafts/mirafedas/twp#mini-plans-web-cta-creative-cloud-card

With class authored page - second one: in list has problem in Devices - Mobile tabelet where the Header and footer of modal -iframe is not sticky. which results in long gap as well below the card
image

Plz take a look, thanks

**Comment by Snehayt** [comment]
Hi @mirafedas ,
Verified the given fix in CC, the continue button in iphone and ipad is hidden.
https://main--cc--adobecom.hlx.live/de/creativecloud/animation/discover?martech=off&milolibs=mwpw-142798-autoset-height-fit-content--milo--mirafedas#animate
image

**Comment by mirafedas** [comment]
@Snehayt and @Roycethan 
thank you, I fixed the above issues, could you please check again?

I created a sandbox page with different modal variations, just sharing, maybe it will be useful: 
https://mwpw-142798-autoset-height-fit-content--milo--mirafedas.hlx.live/drafts/mirafedas/twp

**Comment by Roycethan** [comment]
@mirafedas  Overall looks good now, issues which i noted fixed, but i would like to test more on stage post merge to test on more real devices. Thank you

**Comment by Snehayt** [comment]
@mirafedas ,
Noticed an issue specific to **firefox** browser.The content of iframe is not fully loaded even without ur PR but after PR the iframe is not seen loaded in firefox browser.

Without your fix-https://main--cc--adobecom.hlx.live/de/creativecloud/animation/discover#animate
image

With your fix-https://main--cc--adobecom.hlx.live/de/creativecloud/animation/discover?milolibs=mwpw-142798-autoset-height-fit-content--milo--mirafedas
image


I also see this issue in homepage-https://main--homepage--adobecom.hlx.live/homepage/index-loggedout#mini-plans-web-cta-photoshop-card for **firefox** browser

**Comment by mirafedas** [comment]
@Snehayt I cannot reproduce this in Firefox, could you please clear the page cache and check again?

**Comment by Snehayt** [comment]
@mirafedas ,looks like the issue in firefox incognito is due to CORS issue.The iframe is fine in firefox browser( normal mode-without incognito).

image
image

Verified the fix in windows in the browsers(chrome,firefox,brave,edge) and the devices(android,iphone,ipad). Looks fine from CC side.

**Comment by mokimo** [comment]
Closes https://github.com/adobecom/milo/pull/1911

**Comment by mirafedas** [comment]
@Snehayt and @Roycethan
Thank you for testing this :) Could you please put the 'verified' label on the PR?

**Comment by mirafedas** [comment]
Converted to a draft after the conversation with Ilyas, rewriting the logic.

File: libs/blocks/modal/modal.css [hunk 1]
```diff
@@ -203,7 +198,11 @@
   .dialog-modal.upgrade-flow-modal {
     border-radius: 0;
   }
-  
+
+  .dialog-modal.commerce-frame .fragment,
```
Comment (yesil) [inline libs/blocks/modal/modal.css:202]: This rule applies to all the nested fragments. Can you make them more explicit ?
  .dialog-modal.commerce-frame > .fragment,
.dialog-modal.commerce-frame > .section {

**Review by yesil — APPROVED** [review]

File: libs/blocks/modal/modal.merch.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    adjustModalHeight(data?.contentHeight);
+  }
+}
+
+export function adjustStyles({ dialog, iframe }) {
+  const isAutoHeightAdjustment = /\/mini-plans\/.*mid=ft.*web=1/.test(iframe.src); // matches e.g. https://www.adobe.com/mini-plans/photoshop.html?mid=ft&web=1
+  if (isAutoHeightAdjustment) {
+    dialog.classList.add('height-fit-content');
+  } else {
+    iframe.style.height = '100%';
```
Comment (Axelcureno) [inline libs/blocks/modal/modal.merch.js:63]: Just curious, in what scenario could this happen?

Comment (mirafedas) [reply libs/blocks/modal/modal.merch.js:63]: @Axelcureno automatic height adjustment happens in Trial with Purchase modal, for example: https://www.adobe.com/products/catalog.html#mini-plans-web-cta-creative-cloud-card

**Comment by Snehayt** [comment]
@mirafedas I see the iframe loading blank again with the fix provided- https://main--cc--adobecom.hlx.live/de/creativecloud/animation/discover?milolibs=mwpw-142798-autoset-height-fit-content--milo--mirafedas
image

**Comment by mirafedas** [comment]
@Snehayt thanks, fixed, please check again

**Pull request merged**